src/main/java/paggro/command/DeleteCommand.java

```diff
@@ -0,0 +1,75 @@
+package paggro.command;
+
+import java.io.IOException;
+
+import paggro.exception.PaggroException;
+import paggro.lister.Lister;
+import paggro.notabledate.NotableDate;
+import paggro.storage.Storage;
+import paggro.task.Deadline;
+import paggro.task.Event;
+import paggro.task.Task;
+import paggro.ui.Ui;
+
+/**
+ * This class encapsulates a delete command which removes a task entry from the list.
+ */
+public class DeleteCommand extends Command {
+    /**
+     * Constructor of DeleteCommand.
+     * @param parameters String of index to be deleted
+     */
+    public DeleteCommand(String parameters) {
+        super(parameters);
+    }
+
+    /**
+     * Carries out the execution of a delete command which removes a task from the list.
+     * @param lister The Lister object for the command to execute on.
+     * @param ui The Ui object for the command to execute on.
+     * @param storage The Storage object for the command to execute on.
+     * @return String of response to the command.
+     * @throws PaggroException
+     */
+    @Override
+    public String execute(Lister lister, Ui ui, Storage storage) throws PaggroException {
+        int index;
+        try {
+            index = Integer.parseInt(this.getParameters());
+        } catch (NumberFormatException e) { // parameter was not a number
+            throw new PaggroException("    Really? Can you input an actual number this time... =.=");
```

Comment:
Perhaps you could abstract the spacings/ tabs out as a variable so that you do not need to manually put in the spacings whenever you output an error message.